core/shared/src/main/scala/zio/Schedule.scala

```diff
@@ -823,20 +823,20 @@ object Schedule {
    * </pre>
    */
   def fixed(interval: Duration): Schedule[Clock, Any, Int] = interval match {
-    case Duration.Infinity                    => once >>> never.as(1)
-    case Duration.Finite(nanos) if nanos == 0 => forever
-    case Duration.Finite(nanos) =>
+    case Duration.Infinity => once >>> never.as(1)
+    case Duration.Zero     => forever
+    case duration =>
```

Comment:
Through `object Finite` apply & unapply, we can preserve backward compatibility here and reduce changes.

Reply:
Oh I see. We still want the notion of Finite. I'll make the changes.

Reply:
I think if we can make this change backward compatible, it would go a long way to easing migration for 1.0. Thank you!